pkg/cmd/server/api/helpers.go

```diff
@@ -120,6 +121,10 @@ func GetMasterFileReferences(config *MasterConfig) []*string {
 	refs = append(refs, &config.ServingInfo.ServerCert.CertFile)
 	refs = append(refs, &config.ServingInfo.ServerCert.KeyFile)
 	refs = append(refs, &config.ServingInfo.ClientCA)
+	for i := range config.ServingInfo.NamedCertificates {
```

Comment:
This is causing is problems.  Embedding in `ServingInfo` makes sense, but now you're trying to chase all `ServingInfo` fields in the config and you've missed some.  I agree that it would be crazy to attach this to say `EtcdConfig`, but it could be there anyway.  Maybe a reflective recurse would be better?